src/core/impl/TChem_Impl_MESA.hpp

```diff
@@ -0,0 +1,1891 @@
+    real_type_1d_dual_view b_zsr; // ("b_zsr", nelectrolyte)
+
+    // local aerosol ions
+    // cations
+    const ordinal_type jc_h   = 0;
+    const ordinal_type jc_nh4 = 1;
+    const ordinal_type jc_na  = 2;
+    const ordinal_type jc_ca  = 3;
+
+    // anions
+    const ordinal_type ja_hso4 = 0;
+    const ordinal_type ja_so4  = 1;
+    const ordinal_type ja_no3  = 2;
+    const ordinal_type ja_cl   = 3;
+    const ordinal_type ja_msa  = 4;
+    // const ordinal_type ja_co3  = 6; (note: this is commented out in MOSAIC as well)
+
+    // MESA parameters
+    const ordinal_type Nmax_MESA = 80;
+    const real_type rtol_mesa = 0.01;
+
+    // Temperature-dependent thermodynamic parameters
+    // liquid-liquid
+    // const ordinal_type nrxn_aer_ll = 3;
+    real_type_1d_dual_view Keq_298_ll; // ("Keq_298_ll", nrxn_aer_ll);
+    real_type_1d_dual_view Keq_a_ll; // ("Keq_a_ll", nrxn_aer_ll);
+    real_type_1d_dual_view Keq_b_ll; // ("Keq_b_ll", nrxn_aer_ll);
+
+    // parameters for MTEM mixing rule (Zaveri, Easter, and Wexler, 2005)
+    real_type_3d_dual_view b_mtem; // ("b_mtem", 6, nelectrolyte, nelectrolyte);
+
+    // molecular weights for aerosol species
+    real_type_1d_dual_view mw_aer_mac;
+
+    // densities for aerosol species
+    real_type_1d_dual_view dens_aer_mac;
+
+    // molecular weights for electrolytes
+    real_type_1d_dual_view mw_electrolyte;
+
+    // magnitude of charges on ions
+    ordinal_type_1d_dual_view zc;
+    ordinal_type_1d_dual_view za;
+
+    // molecular weights of ions [g/mol]
+    real_type_1d_dual_view MW_c;
+    real_type_1d_dual_view MW_a;
+};
+
+template<typename DeviceType>
+struct MOSAIC_ConstData {
+public:
+  using device_type = DeviceType;
+
+  using real_type_1d_view_type = Tines::value_type_1d_view<real_type, device_type>;
+  using real_type_2d_view_type = Tines::value_type_2d_view<real_type, device_type>;
+  using real_type_3d_view_type = Tines::value_type_3d_view<real_type, device_type>;
+
+  using ordinal_type_1d_view_type = Tines::value_type_1d_view<ordinal_type, device_type>;
+
+  using mcd_real_type_1d_view = ConstUnmanaged<real_type_1d_view_type>;
+  using mcd_real_type_2d_view = ConstUnmanaged<real_type_2d_view_type>;
+  using mcd_real_type_3d_view = ConstUnmanaged<real_type_3d_view_type>;
+
+  using mcd_ordinal_type_1d_view = ConstUnmanaged<ordinal_type_1d_view_type>;
+
+  mcd_real_type_2d_view a_zsr;
+  mcd_real_type_1d_view aw_min;
+  mcd_real_type_1d_view b_zsr;
+  mcd_real_type_1d_view Keq_298_ll;
+  mcd_real_type_1d_view Keq_a_ll;
+  mcd_real_type_1d_view Keq_b_ll;
+  mcd_real_type_3d_view b_mtem;
+  mcd_real_type_1d_view mw_aer_mac;
+  mcd_real_type_1d_view dens_aer_mac;
+  mcd_real_type_1d_view mw_electrolyte;
+  mcd_ordinal_type_1d_view za, zc;
+  mcd_real_type_1d_view MW_c, MW_a;
+
+
+  // Constructor for MOSAIC constants non-constant arrays of known size
+  MOSAIC_ConstData(const MosaicModelData &mmd) {
+
+    a_zsr(0, mmd.jnh4hso4) =   1.30894;
```

Comment:
We need to set these values in a host view. Then, we should sync them to the device. For example, use a dual_view

a_zsr = real_type_2d_dual_view(do_not_init_tag("TChem-Mosaic::a_zsr"), d1, d2); // where d1 and d2 are the sizes of this view.
// Get host view
auto a_zsr_host = a_zsr.view_host();
// Set values
a_zsr_host(0, mmd.jnh4hso4) = 1.30894;
// ...
// Sync to device
a_zsr.modify_host();
a_zsr.sync_device();